### PR TITLE
Show text info about next turn street/road longer

### PR DESCRIPTION
**Pull request by AntonM030481**
Text info about next turn street/road will be displayed
at least during 60 seconds before turn.

Fix for #2640

### CODE REVIEW

File: routing/route.cpp [hunk 1]
```diff
@@ -87,34 +87,66 @@ double Route::GetTotalTimeSec() const
...
+  return GetCurrentTimeToSegmentSec(m_routeSegments.size() - 1);
+}
+
+double Route::GetCurrentTimeToNearestTurnSec() const
+{
+  double distance;
+  TurnItem turn;
+  GetNearestTurn(distance, turn);
+
+  // |curTurn.m_index| - 1 is the index of |curTurn| segment.
```
Comment (vng) [inline routing/route.cpp:100]: replace "curTurn" with "turn"

File: routing/routing_session.cpp [hunk 2]
```diff
@@ -436,8 +435,10 @@ void RoutingSession::GetRouteFollowingInfo(FollowingInfo & info) const
 
   info.m_completionPercent = GetCompletionPercent();
 
+  double timeToNearestTurnSec = m_route->GetCurrentTimeToNearestTurnSec();
```
Comment (vng) [inline routing/routing_session.cpp:438]: double const

File: routing/routing_tests/route_tests.cpp [hunk 3]
```diff
@@ -176,52 +162,84 @@ UNIT_TEST(DistanceToCurrentTurnTest)
...
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = 0;
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
```
Comment (vng) [inline routing/routing_tests/route_tests.cpp:177]: TEST(..., (time)); prints time value id test failed.

File: routing/routing_tests/route_tests.cpp [hunk 4]
```diff
@@ -176,52 +162,84 @@ UNIT_TEST(DistanceToCurrentTurnTest)
...
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = (kTestTimes[1 - 1] + kTestTimes[2 - 1]) / 2;
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
```
Comment (vng) [inline routing/routing_tests/route_tests.cpp:196]: TEST(..., (time, timePassed)); prints time value id test failed.

File: routing/routing_tests/route_tests.cpp [hunk 5]
```diff
@@ -176,52 +162,84 @@ UNIT_TEST(DistanceToCurrentTurnTest)
...
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = 0.8 * kTestTimes[2 - 1] + 0.2 * kTestTimes[3 - 1];
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
```
Comment (vng) [inline routing/routing_tests/route_tests.cpp:215]: TEST(..., (time, timePassed)); prints time value id test failed.

File: routing/routing_tests/route_tests.cpp [hunk 6]
```diff
@@ -176,52 +162,84 @@ UNIT_TEST(DistanceToCurrentTurnTest)
...
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = 0.3 * kTestTimes[3 - 1] + 0.7 * kTestTimes[4 - 1];
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
```
Comment (vng) [inline routing/routing_tests/route_tests.cpp:234]: TEST(..., (time, timePassed)); prints time value id test failed.

File: routing/routing_tests/route_tests.cpp [hunk 7]
```diff
@@ -176,52 +162,84 @@ UNIT_TEST(DistanceToCurrentTurnTest)
...
     route.MoveIterator(GetGps(pos.x, pos.y));
 
-    route.GetCurrentTurn(distance, turn);
+    route.GetNearestTurn(distance, turn);
     size_t currentTurnIndex = 2;
     TEST(base::AlmostEqualAbs(distance,
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = (kTestTimes[1 - 1] + kTestTimes[2 - 1]) / 2;
```
Comment (vng) [inline routing/routing_tests/route_tests.cpp:190]: double const here and below.

**Comment by vng** [comment]
Please, clarify in comments that GetNearestTurn returns "nearest turn forward, not backward".

File: routing/route.cpp (hunk 1 above)
Comment (AntonM030481) [reply routing/route.cpp:100]: OK

File: routing/routing_session.cpp (hunk 2 above)
Comment (AntonM030481) [reply routing/routing_session.cpp:438]: OK

File: routing/routing_tests/route_tests.cpp (hunk 7 above)
Comment (AntonM030481) [reply routing/routing_tests/route_tests.cpp:190]: OK

File: routing/routing_tests/route_tests.cpp (hunk 4 above)
Comment (AntonM030481) [reply routing/routing_tests/route_tests.cpp:196]: Changed to TEST_ALMOST_EQUAL_ABS.

File: routing/routing_tests/route_tests.cpp (hunk 5 above)
Comment (AntonM030481) [reply routing/routing_tests/route_tests.cpp:215]: Changed to TEST_ALMOST_EQUAL_ABS.

File: routing/routing_tests/route_tests.cpp (hunk 6 above)
Comment (AntonM030481) [reply routing/routing_tests/route_tests.cpp:234]: Changed to TEST_ALMOST_EQUAL_ABS.

**Pull request merged**